lib/licensed/sources/yarn.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+require "json"
+
+module Licensed
+  module Sources
+    class Yarn < Source
+      def enabled?
+        return unless Licensed::Shell.tool_available?("yarn") && Licensed::Shell.tool_available?("npm")
+
+        config.pwd.join("package.json").exist? && config.pwd.join("yarn.lock").exist?
+      end
+
+      def enumerate_dependencies
+        packages.map do |name, package|
+          Dependency.new(
+            name: name,
+            version: package["version"],
+            path: package["path"],
+            metadata: {
+              "type"     => Yarn.type,
+              "name"     => package["name"],
+              "summary"  => package["description"],
+              "homepage" => package["homepage"]
+            }
+          )
+        end
+      end
+
+      def packages
+        root_dependencies = JSON.parse(yarn_list_command)["data"]["trees"]
```

Comment:
Should this check for `"type": "tree"`? For example `yarn list` for me produces two additional lines on stderr:

```
{"type":"warning","data":"package.json: License should be a valid SPDX license expression"}
{"type":"warning","data":"<removed>@1.0.0: License should be a valid SPDX license expression"}
```

Reply:
ah interesting, yup will add 👍